Update "Custom rules and decoders" documentation (#5954)
wazuh/wazuh-documentation: source/user-manual/ruleset/custom.rst

              full event: 'Dec 25 20:45:02 MyHost example[12345]: User 'admin' logged from '192.168.1.100''
              timestamp: 'Dec 25 20:45:02'
              hostname: 'MyHost'
              program_name: 'example'

      **Phase 2: Completed decoding.
              name: 'example'
              dstuser: 'admin'
              srcip: '192.168.1.100'

      **Phase 3: Completed filtering (rules).
              id: '100010'
              level: '0'
              description: 'User logged'
              groups: '['local', 'syslog', 'sshd']'
              firedtimes: '1'
              mail: 'False'

   To test your rules and decoders using ``wazuh-logtest``, it's enough to save the changes. However, you need to restart the Wazuh manager to generate alerts based on these changes.  

#. Restart the Wazuh manager to load the updated rules and decoders:

      .. include:: /_templates/installations/manager/restart_wazuh_manager.rst

Changing an existing rule
---------------------------

.. warning::
    Changes to any rule file inside the ``/var/ossec/ruleset/rules`` folder will be lost in the update process. Use the following procedure to preserve your changes.

> You can change the default Wazuh rules. To do so, we recommend copying the rules to a file in the ``/var/ossec/etc/rules/`` directory, making the necessary changes, and adding the ``overwrite="yes"`` tag to the modified rules. These steps guarantee that your changes won't be lost during updates.

Reviewer: ```suggestion
You can change the default Wazuh rules. We recommend three general steps for this:

#. Copying the rules to a file in the ``/var/ossec/etc/rules/`` directory.
#. Making the necessary changes.
#. Adding the ``overwrite="yes"`` tag to the modified rules.

These steps guarantee that your changes won't be lost during updates.
```

> Here's an example of to change the level value of the SSH rule ``5710`` from 5 to 10.

Reviewer: ```suggestion
Here's an example on how to change the level value of the SSH rule ``5710`` from 5 to 10.
```

#. Open the rule file ``/var/ossec/ruleset/rules/0095-sshd_rules.xml``.

> #. Find and copy the following code from the rule file:

Reviewer: ```suggestion
#. Find and copy the rule definition for rule id ``5710``. It's similar to this:
```

        <group>invalid_login,authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,pci_dss_10.6.1,gpg13_7.1,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
      </rule>

> #. Paste the code into ``/var/ossec/etc/rules/local_rules.xml``, modify the level value, and add ``overwrite="yes"`` to indicate that this rule is overwriting an already defined rule:

Reviewer: ```suggestion
#. Paste the copied rule definition into ``/var/ossec/etc/rules/local_rules.xml``. Modify the level value, and add ``overwrite="yes"`` to indicate that this rule is overwriting an already defined rule:
```

        <group>invalid_login,authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,pci_dss_10.6.1,gpg13_7.1,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
      </rule>

   .. warning::
      To maintain consistency between loaded rules, currently it's not possible to overwrite the ``if_sid``, ``if_group``, ``if_level``, ``if_matched_sid``, and ``if_matched_group`` labels. These tags are ignored when they are in an overwrite rule, keeping the original values.

#. Restart the Wazuh manager to load the updated rules:

   .. include:: /_templates/installations/manager/restart_wazuh_manager.rst

Changing an existing decoder
-----------------------------

.. warning::

>     Changes in any decoder file in the ``/var/ossec/ruleset/decoders`` folder will be lost in the update process. Use the following procedure to preserve your changes.

Reviewer: ```suggestion
    Changes in any decoder file in the ``/var/ossec/ruleset/decoders`` folder are lost in the update process. Use the following procedure to preserve your changes.
```

> To change a default decoder, you can rewrite its file in the ``/var/ossec/etc/decoders`` directory, make the changes, and exclude the original decoder file from the loading list. 

Reviewer: ```suggestion
To change a default decoder, you can:

#. Place a copy of its file in the ``/var/ossec/etc/decoders`` directory.
#. Make changes.
#. Exclude the original decoder file from the loading list.
```

> If you want to customize the decoder file ``0310-ssh_decoders.xml``, follow these steps: 

Reviewer: ```suggestion
For example, if you want to customize decoders in the file ``0310-ssh_decoders.xml``, follow these steps: 
```

> #. Copy the decoder file ``/var/ossec/ruleset/decoders/0310-ssh_decoders.xml`` from the default folder to the user folder ``/var/ossec/etc/decoders`` in order to keep the changes.

Reviewer: ```suggestion
#. Copy the decoder file ``/var/ossec/ruleset/decoders/0310-ssh_decoders.xml`` to the user folder ``/var/ossec/etc/decoders``. This keeps the changes you make when updating to a newer version.
```

> #. Use the ``<decoder_exclude>`` tag  in the ``/var/ossec/etc/ossec.conf`` configuration file to exclude the original decoder file ``ruleset/decoders/0310-ssh_decoders.xml`` from the loading list. Thus, the specified decoder will not be loaded from the default decoder folder, and the decoder file saved in the user folder will be loaded instead.

Reviewer: ```suggestion
#. Edit the ``/var/ossec/etc/ossec.conf`` configuration file. Set the ``<decoder_exclude>`` tag to exclude the original ``ruleset/decoders/0310-ssh_decoders.xml`` decoder file from the loading list. With this configuration, Wazuh loads the decoder file located in the user folder and not the file in the default folder.
```

      </ruleset>

> #. Perform the changes in the file ``/var/ossec/etc/decoders/0310-ssh_decoders.xml``.

Reviewer: ```suggestion
#. Make changes to ``/var/ossec/etc/decoders/0310-ssh_decoders.xml``.
```

#. Restart the Wazuh manager so the changes take effect:

   .. include:: /_templates/installations/manager/restart_wazuh_manager.rst

   .. warning::

>       Note that at this point, if updates to the public Wazuh ruleset include changes to ``0310-ssh_decoders.xml``, they will not apply to you since you are no longer loading that decoder file from the standard location that gets updates.  At some point, you may have to manually migrate your customized material from ``0310-ssh_decoders.xml`` to a newer copy of that file.  Consider internally documenting your changes in ``0310-ssh_decoders.xml`` so that they are easy to find if they have to be migrated later.

Reviewer: ```suggestion
      Note that, since you are excluding the original decoder, you aren't benefiting from any updates it might get. You're no longer loading that decoder file from the standard location that gets the update.  Thus, you need to repeat your customization using the newer default copy of the file in this case.  Consider documenting your changes so that they are easy to find if you need to migrate them later.
```
